zip-0400.rst

```diff
@@ -0,0 +1,174 @@
+::
+
+  ZIP: 400
+  Title: Wallet.dat format
+  Owners: Alfredo Garcia <oxarbitrage@gmail.com>
+  Status: Draft
+  Category: Wallet
+  Created: 2020-05-26
+  License: MIT
+
+
+Terminology
+===========
+
+The key words "MUST" and "MAY" in this document are to be interpreted as described in
+RFC 2119. [#RFC2119]_
+
+Abstract
+========
+
+This proposal defines the current zcash format for wallet persistence storage commonly known as wallet.dat.
+
+Motivation
+==========
+
+The process of saving wallet data into disk is currently unspecified. The key-values used in the current implementation are undocumented and by this structure and functionality is unknown without a deep analysis of the involved source code. This document details the schema and the mechanics of the wallet database. This is an informational document, no changes will be done to the source code.
+
+Specification
+=============
+
+Zcash stores wallet information in a Berkeley database(BDB) commonly known as wallet.dat. 
+The main purpose of this is the persistence of public and private keys the user created and the ability to recover the state after a node restart. This file also allows the migration of user information from one node to another by moving the database to the corresponding new datadir assuming both zcashd instances are running the same or similar version. Additional features like encryption, recovery, etc are also implemented.
+
+The current database is a key-value store where keys and values can have multiple data types in binary format. The first data found in a database key is always the property name, the rest of the key is generally used to identify a record, for example::
+
+  <------ KEY ----+- VALUE ->
+  ---------------------------
+  | zkey | pubkey | privkey |
+  ---------------------------
+  
+Here ``zkey`` is the property name located at the first position of the database key, the public key is also part of the database key and it is located in the second position of it, the private key is saved in the database value column at the first position. More examples opf this can be found at [#ZIP400Attempt]_.
+
+Schema
+------
+
+According to Zcash v3.0.0-rc1 the following key-values can be found, the property names in **bold** means only 1 of this type can exist in the entire database while the others can be found duplicated. Keys and Values columns of the table contains the types the stored data is representing. Included also there are the variable names hoping it will add some clarity to what the stored data is representing.
```

Comment:
```suggestion
According to Zcash v3.0.0-rc1 the following key-values can be found: the property names in **bold** mean only one instance of this type can exist in the entire database, while the others can have multiple instances. Keys and Values columns of the table contain the types that the stored data is representing. Included also there are the variable names hoping it will add some clarity to what the stored data is representing.
```